engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -780,17 +780,17 @@ public void monitorHeartbeat(boolean waitForRecovery)
           //LOG.debug("{} {} {}", c.getExternalId(), currentTms - sca.createdMillis, this.vars.heartbeatTimeoutMillis);
           // container allocated but process was either not launched or is not able to phone home
           if (currentTms - sca.createdMillis > 2 * this.vars.heartbeatTimeoutMillis) {
-            LOG.error("Container {}@{} startup timeout ({} ms).", c.getExternalId(), c.host, currentTms - sca.createdMillis);
+            LOG.warn("Container {}@{} startup timeout ({} ms).", c.getExternalId(), c.host, currentTms - sca.createdMillis);
             containerStopRequests.put(c.getExternalId(), c.getExternalId());
           }
         } else {
           if (currentTms - sca.lastHeartbeatMillis > this.vars.heartbeatTimeoutMillis) {
             if (!isApplicationIdle()) {
               // Check if the heartbeat for this agent has already been missed to raise the StramEvent only once
               if (sca.lastHeartbeatMillis != -1) {
-                String info = String.format("Container %s@%s heartbeat timeout (%d%n ms).", c.getExternalId(), c.host, currentTms - sca.lastHeartbeatMillis);
-                LOG.error(info);
-                StramEvent stramEvent = new StramEvent.ContainerErrorEvent(c.getExternalId(), info, null);
+                String info = String.format("Container %s@%s heartbeat timeout  (%d%n ms).", c.getExternalId(), c.host, currentTms - sca.lastHeartbeatMillis);
+                LOG.warn(info);
+                StramEvent stramEvent = new StramEvent.ContainerErrorEvent(c.getExternalId(), info, null, StramEvent.LogLevel.WARN);
```

Comment:
The event is not due to an internal error or condition that cannot be handled by the system and hence ERROR level in log4j isn't appropriate. This has been discussed before and we don't use logging at this level for similar conditions elsewhere (at least I hope that is still the case). 

Exception handling: If an exception is "handled", then there is normally no need to also log it (at least not as error). Such logging usually occurs for exceptions that are not handled in a catch all handler, where there is no other action that can be taken.
 
I don't want to go in circles. If we cannot reach agreement then the logger change needs to be reverted as part of the original JIRA.